TestRefreshClient_Stress is unstable

https://github.com/networkservicemesh/sdk/runs/2128531196?check_suite_focus=true
```
--- FAIL: TestRefreshClient_Stress (1.11s)
    --- FAIL: TestRefreshClient_Stress/Durations (0.25s)
        assertion_compare.go:230: 
            	Error Trace:	client_utils_test.go:157
            	            				client_utils_test.go:123
            	            				client_utils_test.go:272
            	            				client_test.go:173
            	            				client_test.go:154
            	Error:      	"80015000" is not less than "50000000"
            	Test:       	TestRefreshClient_Stress/Durations
            	Messages:   	[Duration expired (too slow) delta=%v max=%v 80.015ms 50ms]
FAIL
```

--- a/pkg/networkservice/common/refresh/client_test.go
+++ b/pkg/networkservice/common/refresh/client_test.go
@@ -39,17 +39,57 @@ import (
 )
 
 const (
-	expireTimeout     = 100 * time.Millisecond
+	expireTimeout     = 300 * time.Millisecond
 	eventuallyTimeout = expireTimeout
 	tickTimeout       = 10 * time.Millisecond
-	neverTimeout      = 5 * expireTimeout
+	neverTimeout      = expireTimeout
 	maxDuration       = 100 * time.Hour
 
 	sandboxExpireTimeout = 300 * time.Millisecond
-	sandboxMinDuration   = 5 * time.Millisecond
-	sandboxStepDuration  = 50 * time.Millisecond
-	sandboxTotalTimeout  = 300 * time.Millisecond
+	sandboxMinDuration   = 50 * time.Millisecond
+	sandboxStepDuration  = 10 * time.Millisecond
+	sandboxTotalTimeout  = 800 * time.Millisecond
 )
+
+func TestRefreshClient_ValidRefresh(t *testing.T) {
+	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	cloneClient := &countClient{
+		t: t,
+	}
+	client := chain.NewNetworkServiceClient(
+		serialize.NewClient(),
+		updatepath.NewClient("refresh"),
+		refresh.NewClient(ctx),
+		adapters.NewServerToClient(updatetoken.NewServer(sandbox.GenerateExpiringToken(expireTimeout))),
+		cloneClient,
+	)
+
+	conn, err := client.Request(ctx, &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			Id: "id",
+		},
+	})
+
+	require.NoError(t, err)
+	require.Condition(t, cloneClient.validator(1))
+
+	require.Eventually(t, cloneClient.validator(2), eventuallyTimeout, tickTimeout)
+
+	lastRequestConn := cloneClient.GetLastRequest().GetConnection()
+	require.Equal(t, conn.Id, lastRequestConn.Id)
+	require.Equal(t, len(conn.Path.PathSegments), len(lastRequestConn.Path.PathSegments))
+	for i := 0; i < len(conn.Path.PathSegments); i++ {
+		connSegment := conn.Path.PathSegments[i]
+		lastRequestSegment := lastRequestConn.Path.PathSegments[i]
+		require.Condition(t, func() (success bool) {
+			return connSegment.Expires.AsTime().Before(lastRequestSegment.Expires.AsTime())
+		})
+	}
+}
 
 func TestRefreshClient_StopRefreshAtClose(t *testing.T) {
 	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
@@ -113,7 +153,7 @@ func TestRefreshClient_RestartsRefreshAtAnotherRequest(t *testing.T) {
 	require.Eventually(t, cloneClient.validator(2), eventuallyTimeout, tickTimeout)
 
 	_, err = client.Request(ctx, &networkservice.NetworkServiceRequest{
-		Connection: conn,
+		Connection: conn.Clone(),
 	})
 	require.NoError(t, err)
 
@@ -130,35 +170,17 @@ type stressTestConfig struct {
 	iterations               int
 }
 
-func TestRefreshClient_Stress(t *testing.T) {
-	t.Skip("https://github.com/networkservicemesh/sdk/issues/774")
-
-	table := []stressTestConfig{
-		{
-			name:          "RaceConditions",
-			expireTimeout: 2 * time.Millisecond,
-			minDuration:   0,
-			maxDuration:   maxDuration,
-			tickDuration:  8100 * time.Microsecond,
-			iterations:    100,
-		},
-		{
-			name:          "Durations",
-			expireTimeout: 50 * time.Millisecond,
-			minDuration:   10 * time.Millisecond,
-			maxDuration:   50 * time.Millisecond,
-			tickDuration:  25 * time.Millisecond,
-			iterations:    10,
-		},
-	}
-	for i := range table {
-		it := &table[i]
-		t.Run(it.name, func(t *testing.T) { runStressTest(t, it) })
-	}
-}
-
-func runStressTest(t *testing.T, conf *stressTestConfig) {
+func TestRefreshClient_CheckRaceConditions(t *testing.T) {
 	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
+
+	conf := &stressTestConfig{
+		name:          "RaceConditions",
+		expireTimeout: 2 * time.Millisecond,
+		minDuration:   0,
+		maxDuration:   maxDuration,
+		tickDuration:  8100 * time.Microsecond,
+		iterations:    100,
+	}
 
 	refreshTester := newRefreshTesterServer(t, conf.minDuration, conf.maxDuration)
 

--- a/pkg/networkservice/common/refresh/client_utils_test.go
+++ b/pkg/networkservice/common/refresh/client_utils_test.go
@@ -26,12 +26,13 @@ import (
 	"time"
 
 	"github.com/golang/protobuf/ptypes/empty"
-	"github.com/networkservicemesh/api/pkg/api/networkservice"
-	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/cls"
-	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/cls"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
 
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
@@ -42,8 +43,10 @@ const (
 )
 
 type countClient struct {
-	t     *testing.T
-	count int32
+	t           *testing.T
+	count       int32
+	lastRequest *networkservice.NetworkServiceRequest
+	mutex       sync.Mutex
 }
 
 func (c *countClient) validator(atLeast int32) func() bool {
@@ -67,11 +70,21 @@ func (c *countClient) Request(ctx context.Context, request *networkservice.Netwo
 		assert.Equal(c.t, endpointName, conn.NetworkServiceEndpointName)
 	}
 
+	c.mutex.Lock()
+	c.lastRequest = request.Clone()
+	c.mutex.Unlock()
+
 	return next.Client(ctx).Request(ctx, request, opts...)
 }
 
 func (c *countClient) Close(ctx context.Context, conn *networkservice.Connection, opts ...grpc.CallOption) (*empty.Empty, error) {
 	return next.Client(ctx).Close(ctx, conn, opts...)
+}
+
+func (c *countClient) GetLastRequest() *networkservice.NetworkServiceRequest {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	return c.lastRequest.Clone()
 }
 
 // refreshTestServer is a helper endpoint to check that the Request()/Close()

